File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    from collections import defaultdict
    answer = []
    table = defaultdict(int)
    for i, v in enumerate(name):
        table[v] = yearning[i]
    for arr in photo:
        answer.append(0)
        for i in arr:
            answer[-1] += table[i]
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const pointObj = name.reduce((acc, cur, index) => {
        acc[cur] = yearning[index];
        
        return acc;
    }, {})
    
    return photo.reduce((acc, cur) => {
        acc.push(cur.reduce((point, name) => {
            const currentPoint = pointObj[name];
            if (currentPoint) {
                point += pointObj[name];    
            }
            
            return point;
        }, 0))
        
        return acc;
    }, [])
}